test/sbvr-helper.coffee

```diff
@@ -247,14 +256,56 @@ exports.TableSpace = ->
 							card = nest(quant, ['MaximumCardinality', 'Number'])
 						if card and card[1] is 1
 							bindings = nest(quant, ['AtomicFormulation', 'RoleBinding'], true)[0]
-							if _.some(bindings, (binding) -> getTable(binding[1][1]).table.primitive)
-								@table = 'Attribute'
+							factType = nest(quant, ['AtomicFormulation', 'FactType'])
+							primitiveTable = bindings.map((binding) -> getTable(binding[1][1]).table).find((table) -> table.primitive)
+							baseTable = getTable(factType[1][1]).table
+
+							if baseTable == primitiveTable
+								# If the primitive table is also the base table then it's actually a unique
+								# constraint rather than a cardinality constraint
+								_.find(@baseTable.fields, { fieldName: primitiveTable.name }).index = 'UNIQUE'
+								@matches = _.cloneDeep(@baseTable)
 							else
-								@table = 'ForeignKey'
-							@matches = undefined
+								if primitiveTable
+									@table = 'Attribute'
+								else
+									@table = 'ForeignKey'
+								dataType = primitiveTable?.primitive ? 'ForeignKey'
+
+								@baseTable = baseTable
+
+								required = quant[0] == 'ExactQuantification'
+								fieldName = _(factType).slice(2).map(1).reject((v) -> v == 'has').join('-')
+								field = _.find(@baseTable.fields, { fieldName })
+								if field
+									field.required = required
+								else
+									@baseTable.fields.push({
+										dataType
+										fieldName
+										required: required
+										index: null
+										references:
+											fieldName: if primitiveTable then null else 'id'
+											resourceName: _.last(factType)[1]
+										defaultValue: null
+									})
+								@property = 'tables.' + @baseTable.resourceName
+								@matches = _.cloneDeep(@baseTable)
 					when 'Definition'
-						# Nulling the property just checks that there are no changes to the previous test result.
-						@property = null
+						definition = lf[1]
+						if definition[0] == 'Enum'
```

Comment:
This is the bit that means enum definitions actually get verified, the bit right above is just to allow us to make checks on fields that get added/modified (and with thatI had to make the other extra fields actually have their checks)